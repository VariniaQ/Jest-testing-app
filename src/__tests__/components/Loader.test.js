import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Loader from '../../components/Loader';

describe('Loader', () => {
  it('should render the Loader component', () => {
    const { container } = render(<Loader />);
    const loaderElement = container.querySelector('.loader');

    // Assert that the loader element is in the document
    expect(loaderElement).toBeInTheDocument();

    // Create a snapshot of the Loader component and compare with the saved snapshot
    const component = renderer.create(<Loader />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
