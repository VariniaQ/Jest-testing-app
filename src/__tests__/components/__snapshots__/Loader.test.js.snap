// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render the Loader component 1`] = `
<div
  className="loader"
/>
`;
